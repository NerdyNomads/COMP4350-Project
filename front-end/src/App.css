.App { 
    display: flex;
    flex-direction: row;

    width: 100%;
	height: 100%;
}

.Collaborators{
    display: flex;
    flex-direction: column;
    width:100%;
}
