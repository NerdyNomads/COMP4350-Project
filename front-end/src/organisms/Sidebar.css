.SideBar {
    z-index: 2;

    position: fixed;
    display: flex;
    flex-direction: column;

    background-color: #FFFFFF;
    max-width: 250px;
    min-width: 250px;
    height: 100vh;
}

.SideBar-logo{
    padding: 20px 10px;
}

.SideBar-logo-text{
    text-align: center;
    font-weight: 700;
    font-size: x-large;
}

.SideBar-divider{
    background-color: #acabab;
    border: 2px solid #acabab;
}

.SideBar-divider-light{
    background-color: #cccccc6c;
    border: 1px solid #cccccc6c;
}

.SideBar-title{
    text-align: center;
    font-style: italic;
    font-size: large;
    padding: 10px 5px;
}

.SideBar-spacing{
    padding: 5px;
    flex-grow: 2;
}

.SideBar-option{
    display: flex;
    flex-direction: row;
    padding: 10px 0;
}

.SideBar-option-logo{
    padding-left: 20px;
    user-select: none;
}

.SideBar-option-text{
    text-align: left;
    padding-left: 10px;
}

.SideBar-option:hover{
    background-color: rgb(238, 238, 238);
    cursor: pointer;

}