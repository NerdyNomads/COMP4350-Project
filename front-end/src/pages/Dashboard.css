.Dashboard {
    margin-left: 250px; /* size of sidebar */
    width: 100%; /* Fill the rest of the screen*/
}

.Dashboard-title {
    font-weight: 600;
    font-size: 36px;
    margin: 20px;
    display: flex;
    justify-content: space-between;
}

.Dashboard-title-reload {
    display: flex;
    justify-content: space-between;
    column-gap: 10px;
}