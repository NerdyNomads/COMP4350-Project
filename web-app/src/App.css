.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-body {
  background-color: #FFD89D;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
  width: 10vh;
  position:relative;
}

.App-header {
  background-color: #419462;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

.sidebar {
  background-color: #FFD89D;
  width: 30vw;
  position: absolute;
  left: 5vh;
  top: 5vh;
  height: 30vh;
  color: black;
  border-radius: 5vh;
}

.text-list {
  background-color: #FFD89D;
  width: 60vw;
  position: absolute;
  right: 5vh;
  height: 100vh;
  color: black;
  padding-top: 5vh;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
