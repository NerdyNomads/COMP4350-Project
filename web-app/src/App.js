import React, { useState } from 'react';
import logo from './logo.svg';
import './App.css';
import Profile from "./components/Profile";
import LoginButton from "./components/LoginButton";
import LogoutButton from "./components/LogoutButton";
import { useAuth0 } from '@auth0/auth0-react';

function App() {
    const [data, setData] = React.useState(null);
    const { isLoading } = useAuth0();

    const [email, setEmail] = useState(null);
    const [username, setUsername] = useState(null);
    const [password, setPassword] = useState(null);

    // function handleEmailChange(email) {
    //     setEmail(email);
    // }

    // const handleAccountSubmit = event => {
    //     // Get the data from the form.
    //     // event.target represents the form that is being submitted.
    //     var formData = new FormData(event.target);

    //     const account = new Account;

    //     var output = "";

    //     // Loop over each pair (name and value) in the form and add it to the values array.
    //     for (var pair of formData) {
    //         if (pair[0] == "email")
    //             account.email = pair[1];
    //         else if (pair[0] == "username")
    //             account.username = pair[1];
    //         else if (pair[0] == "password")
    //             account.password = pair[1];
    //     }
    //     output = "Email: " + account.email + " " +
    //         "Username: " + account.username + " " +
    //         "Password: " + account.password;

    //     console.log(output);

    //     event.preventDefault();
    // };

    // Test backend connection using GET route from server.js
    React.useEffect(() => {
        fetch("/api")
            .then((res) => res.json())
            .then((data) => setData(data.message));
    }, []);

    return (
        <div className="App">
            <header className="App-header">
                <p>{!data ? "Loading..." : data}</p>
                <LoginButton />
                <LogoutButton />
                <Profile />
            </header>

            <h1>Account Input Test</h1>
            <form id="account_form">
                <label for="email">Email:</label><br />
                <input type="text" id="email" name="email" /><br />
                <label for="username">User Name:</label><br />
                <input type="text" id="username" name="username" /><br />
                <label for="password">Password:</label><br />
                <input type="text" id="password" name="password" /><br /><br />
                <input type="submit" value="submit"  />
            </form>
            <p id="account_create"></p><br />

            <h1>Workspace Input Test</h1>
            <form id="workspace_form">
                <label for="workspace_name">Name:</label><br />
                <input type="text" id="workspace_name" name="workspace_name" /><br /><br />
                <input type="submit" value="submit" />
            </form>
            <p id="workspace_create"></p><br />

            <h1>Text Content Input Test</h1>
            <form id="text_form">
                <label for="text_content_name">Name:</label><br />
                <input type="text" id="text_content_name" name="text_content_name" /><br />
                <label for="text_value">Text:</label><br />
                <input type="text" id="text_value" name="text_value" /><br />
                <label for="source">Source:</label><br />
                <input type="text" id="source" name="source" /><br /><br />
                <input type="submit" value="submit" />
            </form>
            <p id="text_create"></p><br />
        </div>
    );
}

export default App;
