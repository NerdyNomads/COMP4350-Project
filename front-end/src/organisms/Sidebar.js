import React from "react";

import { useAuth0 } from "@auth0/auth0-react";


import { GearSmallIcon, LogoutIcon } from "../atoms/icons";
import "./Sidebar.css";
import SidebarWorkspace from "../molecules/SidebarWorkspace";


function Sidebar() {

  // FAKE DB FOR UI
  // const workspaces = [
  //   {
  //     id: "1",
  //     name: "Test 1"
  //   },
  //   {
  //     id: "2",
  //     name: "Test 2"
  //   },
  //   {
  //     id: "3",
  //     name: "Test 3"
  //   }
  // ];

  // const texts = [
  //   {
  //     id: "10",
  //     text: "Testing text from Test 1",
  //     source: "https://www.test1.com",
  //     workspaceID: "1",
  //     creationDate: 1646724250020
  //   },
  //   {
  //     id: "20",
  //     text: "Testing text from Test 2",
  //     source: "https://www.test2.com",
  //     workspaceID: "2",
  //     creationDate: 1646760121099,

  //   },
  //   {
  //     id: "30",
  //     text: "Testing text from Test 3",
  //     source: "https://www.test3.com",
  //     workspaceID: "3",
  //     creationDate: 1646767121099
  //   }
  // ];

  const { logout, user } = useAuth0();

  const spacing = <div className="SideBar-spacing"/>;
  const divider = <div className="SideBar-divider"/>;
  const dividerLight = <div className="SideBar-divider-light"/>;
  const componentName = "SideBar";

  const handleManageAccountOnClick = () => {
    alert("Temporary: Manage Account subfeature coming soon.");
  };

  return (
    <div className={`${componentName}`}>
      <div className={`${componentName}-logo`}>
        <div className={`${componentName}-logo-text`}>TextSavvy</div>
      </div>
      {divider}
      {/* eslint-disable-next-line react/no-unescaped-entities */ }
      <div className={`${componentName}-title`}>{user.nickname}'s Workspaces</div>

      {dividerLight}

      <SidebarWorkspace/>

      {dividerLight}
      {spacing}
      <div className={`${componentName}-manageacc ${componentName}-option`} onClick={handleManageAccountOnClick}>
        <div className={`${componentName}-option-logo`}><GearSmallIcon/></div>
        <div className={`${componentName}-option-text`}>Manage Account</div>
      </div>
      <div className={`${componentName}-logout ${componentName}-option`} onClick={logout}>
        <div className={`${componentName}-option-logo`}><LogoutIcon/></div>
        <div className={`${componentName}-option-text`}>Logout</div>
      </div>
      
      
    </div>
  );

}

export default Sidebar;