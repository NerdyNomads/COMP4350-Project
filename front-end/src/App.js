import React, { useEffect, useState } from "react";
import { useAuth0 } from "@auth0/auth0-react";
import axios from "axios";

import SideBar from "./organisms/Sidebar";
import Dashboard from "./pages/Dashboard";
import "./App.css";
import { addNewAccount } from "./util/requests";

function App() {
  const [currentAccountId, setCurrentAccountId] = useState(null);
  const [currentWorkspaceId, setCurrentWorkspaceId] = useState("");
  const { isAuthenticated, user, isLoading, loginWithRedirect, getAccessTokenSilently } = useAuth0();
  
  async function getAccount(auth0Id) {
    let result = await axios.get(`${process.env.REACT_APP_BACKEND_SERVER}/accounts/auth0/` + auth0Id);

    return result?.data[0];
  }

  useEffect(async () => {
    let abortController = new AbortController();
    if(isLoading){
      console.log("app is loading");
      return;
    }

    if(!isAuthenticated){
      console.log("not authenticated. redirecting to login");
      loginWithRedirect();
      return;
    
    // If someone is currently logged in, then this checks if the account needs to be added to the database.
    } else {
      let userId = user.sub;
      let auth0Id = userId.replace("|", "-");	// Must replace "|" to allow for GET query

      let account = await getAccount(auth0Id);

      if (account) {
        setCurrentAccountId(account._id);
      } else {
        const response = await addNewAccount(auth0Id,user);
        setCurrentAccountId(response.data._id);
      }
    }

    getToken({});
    return () => {  
      abortController.abort();  
    }; 
  }, [isLoading, isAuthenticated]);

  const getToken = (options) => {
    console.log(`retrieving token for '${options.audience}'`);
    getAccessTokenSilently(options).then()
      .catch(e => {
        console.error("failed to get token!", e);
      });
  };
  
  const handleGoToWorkspace = (id) => setCurrentWorkspaceId(id);

  if (!isAuthenticated) {
    return <></>;
  } else {
    return (
      <div className="App">
        <SideBar onClickWorkspace={handleGoToWorkspace} accountId={currentAccountId}/>
        <Dashboard workspaceId={currentWorkspaceId}/>
      </div>
    );
  }	
}

export default App;
