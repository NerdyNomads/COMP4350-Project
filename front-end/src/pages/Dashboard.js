import React, {useState, useEffect} from "react";

import axios from "axios";

import TextList from "../molecules/TextList";
import "./Dashboard.css";

function Dashboard() {
  const [ textItems, setTextItems ] = useState(null);

  /**
   * Grabs data from the db.
   * 
   * @returns 				returns data from the db. 
   */
  async function getTexts() {
    let result = await axios.get(`${process.env.REACT_APP_BACKEND_SERVER}/texts`);
    return result?.data;
  }

  useEffect(async () => {
    let texts = await getTexts();

    if (texts) {
      setTextItems(texts);
    }
  }, []);

  return (
    <div className="Dashboard">
      { textItems &&
          <TextList list={textItems}/>
      }
    </div>
  );
}


export default Dashboard;