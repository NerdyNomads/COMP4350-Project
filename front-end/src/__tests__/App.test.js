/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
import React from "react";
import { shallow } from "enzyme";

import App from "../App";
import { useAuth0 } from "@auth0/auth0-react";
import Sidebar from "../organisms/Sidebar";
import Dashboard from "../pages/Dashboard";

jest.mock("@auth0/auth0-react"); 

const auth0User = {
  sub: "abc|12345",
  nickname: "Test"
};

test("Should match the snapshot.", () => { 
  useAuth0.mockReturnValue({
    isAuthenticated: true,
    user: auth0User
  });

  const wrapper = shallow(<App />);
  expect(wrapper.html()).toMatchSnapshot();
});

test("Sidebar and Dashboard should be visible if user is logged in.", () => { 
  useAuth0.mockReturnValue({
    isAuthenticated: true,
    user: auth0User
  });

  const wrapper = shallow(<App />);
  expect(wrapper.containsMatchingElement(<Sidebar/>)).toEqual(true);
  expect(wrapper.containsMatchingElement(<Dashboard/>)).toEqual(true);
});

test("Component should be empty if no user is logged in.", () => { 
  useAuth0.mockReturnValue({
    isAuthenticated: false
  });

  const wrapper = shallow(<App />);
  expect(wrapper.html()).toEqual("");
  expect(wrapper.containsMatchingElement(<Sidebar/>)).toEqual(false);
  expect(wrapper.containsMatchingElement(<Dashboard/>)).toEqual(false);
});

// TODO:
// - loginWithRedirect
// - isNewAccount
// - addNewAccount
