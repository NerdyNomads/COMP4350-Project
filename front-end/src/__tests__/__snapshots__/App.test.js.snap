// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot. 1`] = `"<div class=\\"App\\"><div class=\\"SideBar\\"><div class=\\"SideBar-logo\\"><div class=\\"SideBar-logo-text\\">TextSavvy</div></div><div class=\\"SideBar-divider\\"></div><div class=\\"SideBar-title\\">Test&#x27;s Workspace</div><div class=\\"SideBar-divider-light\\"></div><div class=\\"SideBar-spacing\\"></div><div class=\\"SideBar-manageacc SideBar-option\\"><div class=\\"SideBar-option-logo\\"><img alt=\\"gear\\" src=\\"/assets/images/gear_sm.svg\\" class=\\"icon\\"/></div><div class=\\"SideBar-option-text\\">Manage Account</div></div><div class=\\"SideBar-logout SideBar-option\\"><div class=\\"SideBar-option-logo\\"><img alt=\\"logout\\" src=\\"/assets/images/logout.svg\\" class=\\"icon\\"/></div><div class=\\"SideBar-option-text\\">Logout</div></div></div><div class=\\"Dashboard\\"></div><div class=\\"Collaborators\\"><div class=\\"CollaboratorItem\\"><span class=\\"CollaboratorItem-email\\">Test</span><img alt=\\"trash can\\" src=\\"/assets/images/trash_can.svg\\" class=\\"CollaboratorItem-remove\\"/></div><div class=\\"CollaboratorItem\\"><span style=\\"font-style:italic;color:#707070\\" class=\\"CollaboratorItem-email\\">Pending Text</span><img alt=\\"trash can\\" src=\\"/assets/images/trash_can.svg\\" class=\\"CollaboratorItem-remove\\"/></div></div></div>"`;
